src/components/tools/Stats/StatOptions.svelte

```diff
@@ -39,17 +41,56 @@
     }
   }
 
+  /**
+   * When user types in the year, the combobox dropdown menu is filtered to show only
+   * values that complete the characters typed in the combobox
+   **/
+  function shouldFilterItem(item, value) {
+    if (!value) return true;
+    return item.text.includes(value);
+  }
+
+  /**
+   * When user presses the Enter or Tab key in combobox after typing in a year, update
+   * the combobox selected index if the year matches an item in the combobox dropdown menu
+   **/
+  function updateIndex(e) {
+    const { key, keyCode, target } = e;
+    if (["Enter", "Tab"].includes(key) || [13, 9].includes(keyCode)) {
```

Comment:
I think with modern browsers we can get away with only using `key` and don't need to also check for `keyCode`. Using `keyCode` can be more error prone from my understanding.